connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java

```diff
@@ -448,14 +448,14 @@ public boolean isDeleted() {
             return deleted;
         }
 
-        public boolean canWrite(T status) {
-            return value != null &&
-                    (value.workerId().equals(status.workerId())
-                    || value.generation() <= status.generation());
+        public boolean canWriteSafely(T status) {
+            return value == null
```

Comment:
Though we probably should have caught this in a unit test. Maybe you can add one now?